### DISCUSSION TITLE
Hillshade algorithm showing grids instead

### DISCUSSION THREAD

**Discussion by lawrencenika** [developmentseed/titiler#980] [Q&A]
#### Problem description

Tried with `/cog/tiles/WebMercatorQuad/{z}/{x}/{y}.png?url=${encodeURIComponent(objPath)}&rescale=127.0,236.0&algorithm=hillshade&buffer=3` endpoint and parameters with my current COG data to render on DeckGL map's BitmapLayer. I got the grid looking image tiles instead of terrain looking ones:

High zoom: 
![image](https://github.com/user-attachments/assets/01c5af45-5136-4ca7-b23b-18ad4cf6cefd)

Low zoom:
![image](https://github.com/user-attachments/assets/4f46d704-7b43-4239-ab1c-185dc681b8fe)

#### Expected Output

Expected look on other platforms's hillshade algorithm:
![image](https://github.com/user-attachments/assets/8205a161-8564-4821-a470-2e95d597e1cd)

#### Environment Information

titiler:latest (v0.18.6) run in local docker env on macos
with ENV WORKERS_PER_CORE 1

**Comment by vincentsarago** [comment]
do not rescale the data

when passing the `rescale=127.0,236.0` you'll tell titiler rescale the output image from the algorithm, which in the hillshade case, should already be a valid uint8 dataset 

https://github.com/developmentseed/titiler/blob/main/src/titiler/core/titiler/core/factory.py#L615-L619

**Reply by lawrencenika** [reply]
still same grid when not using rescale at all, could it be because my COG raw image is of float32? https://drive.google.com/file/d/1bZQAvxEBbPMyjkXhNYEVyQQck0dNMFRx/view?usp=sharing

**Reply by lawrencenika** [reply]
@vincentsarago this one too

**Reply by vincentsarago** [reply]
@lawrencenika you're file is not a DEM right? it's a 4 bands dataset of type float

The hillshade algorithm is defined here https://github.com/developmentseed/titiler/blob/725f857c735790e6c7783c4319f61804669b1980/src/titiler/core/titiler/core/algorithm/dem.py#L31-L39

The code is pretty basic and used to work for DEM, I'm not sure what Is wrong with the usage you're trying to do.

you're sharing an expected output, can I ask how you created it?

**Reply by lawrencenika** [reply]
yes my image is not DEM but 4 bands of type float. the created output are shown as images above expected output.

**Comment by vincentsarago** [comment]
Ok I think there is something going on with rio-tiler when we are `oversampling` a file somewhere https://github.com/cogeotiff/rio-tiler/blob/741acc3596d908db1755c82f18e2f4fd3711c49c/rio_tiler/reader.py#L375-L423

I'm not quite sure why and where but I'll open an issue in rio-tiler

**Reply by lawrencenika** [reply]
thanks, awesome!

**Comment by vincentsarago** [comment]
@lawrencenika I'm still looking at this.

I already found one issue which will be fixed in https://github.com/developmentseed/titiler/pull/985 and I'll backport this in a non-breaking version (0.18.7, released later today)

The other main issue comes from rio-tiler, when you're over-zooming your dataset, which for some reason creates the `square` artefacts. For now I don't see immediate fix for this, so it would be best if you don't fetch tiles that are over-zooming you're dataset and stop at the `maxzoom` returned in the tilejson endpoint.

Your map client should be able to `over-zoom` the pre-loaded tiles them self

**Reply by lawrencenika** [reply]
awesome to hear! would the rio-tiler issue solved later avoid having to use the tilejson endpoint?

**Reply by vincentsarago** [reply]
I don't really have the bandwidth at the moment. And it's a good habit not to over zoom a dataset anyway!

**Reply by lawrencenika** [reply]
got it thanks!
